src/renderer/entities/network/lib/networkService.ts

```diff
@@ -1,388 +1,115 @@
-          isNetworkMatch = chainId === api.genesisHash.toHex();
-
-          api.disconnect().catch(console.warn);
-        } catch (error) {
-          console.warn(error);
-        }
-        provider.disconnect().catch(console.warn);
-        resolve(isNetworkMatch ? RpcValidation.VALID : RpcValidation.WRONG_NETWORK);
-      });
-
-      provider.on('error', async () => {
-        try {
-          await provider.disconnect();
-        } catch (error) {
-          console.warn(error);
-        }
-        resolve(RpcValidation.INVALID);
-      });
-    });
-  };
-
-  const addRpcNode = async (chainId: ChainId, rpcNode: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
-
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).concat(rpcNode),
-    });
-  };
-
-  const updateRpcNode = async (chainId: ChainId, oldNode: RpcNode, newRpc: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
-
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).map((node) => {
-        if (node.url === oldNode.url && node.name === oldNode.name) return newRpc;
-
-        return node;
-      }),
-    });
-  };
-
-  const removeRpcNode = async (chainId: ChainId, rpcNode: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
+    return new CachedScProvider(Sc, knownChainId);
+  } else {
+    throw new Error('Parachains do not support Substrate Connect yet');
+  }
+};
 
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).filter((node) => node.url !== rpcNode.url),
-    });
-  };
+const createWebsocketProvider = (
+  rpcUrls: string[],
+  chainId: ChainId,
+  getMetadata: (chainId: ChainId) => Promise<Metadata | undefined>,
+): ProviderInterface => {
+  // TODO: handle limited retries provider = new WsProvider(node.address, 5000, {1}, 11000);
+  const CachedWsProvider = createCachedProvider(WsProvider, chainId, getMetadata);
 
-  const setupConnections = async (): Promise<void> => {
-    try {
-      const chainsData = chainsService.getChainsData();
-      chains.current = keyBy(chainsService.sortChains(chainsData), 'chainId');
+  return new CachedWsProvider(rpcUrls, 2000);
+};
 
-      const newConnections = await getNewConnections();
-      await clearConnections();
-      await addConnections(newConnections);
+async function disconnect(provider?: ProviderInterface, api?: ApiPromise) {
+  try {
+    await api?.disconnect();
+  } catch (e) {
+    console.warn(e);
+  }
 
-      const connectionsMap = await getExtendConnections();
-      setConnections(connectionsMap);
-    } catch (error) {
-      console.error(error);
-    }
-  };
+  try {
+    await provider?.disconnect();
+  } catch (e) {
+    console.warn(e);
+  }
```

Comment:
Do we need to call disconnect on bath `ApiPromise` and `ProviderInterface`? Doesnt `ApiPromise.disconnect()` also disconnect its ProviderInterface?